support/ebpf/native_stack_trace.ebpf.c

```diff
@@ -643,6 +643,90 @@ int native_tracer_entry(struct bpf_perf_event_data *ctx)
   }
 
   u64 ts = bpf_ktime_get_ns();
-  return collect_trace((struct pt_regs *)&ctx->regs, TRACE_SAMPLING, pid, tid, ts, 0);
+  return collect_trace(regs, origin, pid, tid, ts, 0, cudaId);
+}
+
+SEC("perf_event/native_tracer_entry")
+int native_tracer_entry(struct bpf_perf_event_data *ctx)
+{
+  return native_tracer_entry_inner((struct pt_regs *)&ctx->regs, TRACE_SAMPLING, 0);
 }
 MULTI_USE_FUNC(unwind_native)
+
+SEC("uprobe/cuda_launch_shim")
+int cuda_launch_shim(struct pt_regs *ctx)
+{
+  u64 pid_tgid = bpf_get_current_pid_tgid();
+  u32 pid      = pid_tgid >> 32;
+  u32 tid      = pid_tgid & 0xFFFFFFFF;
+
+  if (pid == 0 || tid == 0) {
+    return 0;
+  }
+
+  u64 arg = ctx->
+#if defined(__x86_64)
+            di
+#elif defined(__aarch64__)
+            regs[0]
+#else
+  #error "Unsupported architecture"
+#endif
+    ;
+
+  DEBUG_PRINT("btv: attached, func is 0x%llx", arg);
```

Comment:
```suggestion
  DEBUG_PRINT("cuda_launch_shim: func is 0x%llx", arg);
```

left my initials in here